harness/determined/core/_checkpoint.py

```diff
@@ -213,6 +220,12 @@ def delete(self, storage_id: str) -> None:
         """
         self._storage_manager.delete(storage_id)
 
+    def _write_metadata_file(self, ckpt_dir: str, metadata: Optional[Dict[str, Any]]) -> None:
```

Comment:
readability nit: disallow `metadata=None` and leave the defaulting behavior to the calling code, or rename this to `_maybe_write_metadata_file` (which seems less desirable to me).

IMO, a function called write_some_file should always write that file.